js/indicators/tema.src.js

```diff
@@ -0,0 +1,235 @@
+                      SMA
+                    )[1];
+                    emaEMAArr.push(emaEMA);
+                    calDEMA = emaEMA;
+                    // Summing first period points for ema(ema(ema))
+                    if (i < tripledPeriod) {
+                        sum += emaEMA;
+                    }
+                }
+                // Calculate tema
+                if (i >= (tripledPeriod)) {
+                    // First tema point
+                    if (i === (tripledPeriod)) {
+                        SMA = sum / period;
+                    }
+                    ema = EmaArr[i - period - 2];
+                    emaEMA = emaEMAArr[i - 2 * period - 1];
+                    emaEMAEMA = EMA.prototype.calculateEma(
+                      [],
+                      [emaEMA],
+                      1,
+                      EMApercent,
+                      calTEMA,
+                      -1,
+                      SMA
+                    )[1];
+                    TEMAPoint = [
+                        xVal[i - 3],
+                        correctFloat(3 * ema - 3 * emaEMA + emaEMAEMA)
+                    ];
+                    TEMA.push(TEMAPoint);
+                    xDataTema.push(TEMAPoint[0]);
+                    yDataTema.push(TEMAPoint[1]);
+                    calTEMA = emaEMAEMA;
+                }
+            }
+
+            // Calculate the last ema and emaEMA points
+            ema = EmaArr[i - period - 1];
+            emaEMA = EMA.prototype.calculateEma(
+              [],
+              [ema],
+              1,
+              EMApercent,
+              calDEMA,
+              -1,
+              SMA
+            )[1];
+            emaEMAArr.push(emaEMA);
+
+            // Calculate the last tema points
+            ema = EmaArr[i - period - 2];
+            emaEMA = emaEMAArr[i - 2 * period - 1];
+            emaEMAEMA = EMA.prototype.calculateEma(
+              [],
+              [emaEMA],
+              1,
+              EMApercent,
+              calTEMA,
+              -1,
+              SMA
+            )[1];
+            TEMAPoint = [
+                xVal[i - 3],
+                correctFloat(3 * ema - 3 * emaEMA + emaEMAEMA)
+            ];
+            TEMA.push(TEMAPoint);
+            xDataTema.push(TEMAPoint[0]);
+            yDataTema.push(TEMAPoint[1]);
+            calTEMA = emaEMAEMA;
+
+            i++;
+
+            ema = EmaArr[i - period - 2];
+            emaEMA = emaEMAArr[i - 2 * period - 1];
+            emaEMAEMA = EMA.prototype.calculateEma(
+              [],
+              [emaEMA],
+              1,
+              EMApercent,
+              calTEMA,
+              -1,
+              SMA
+            )[1];
+            TEMAPoint = [
+                xVal[i - 3],
+                correctFloat(3 * ema - 3 * emaEMA + emaEMAEMA)
+            ];
+            TEMA.push(TEMAPoint);
+            xDataTema.push(TEMAPoint[0]);
+            yDataTema.push(TEMAPoint[1]);
```

Comment:
Lines 165-182 are the same. Why these points are not in a loop? I see that formula is the same, only `i` is incremented.